### PR TITLE
EA Giving season pre-votes

### PR DESCRIPTION
**Pull request by oetherington**
This PR implements basic pre-voting for giving season election candidates. [Figma](https://www.figma.com/file/3WOTZHH10ZBXSZPWw9omM3/Giving-Season?node-id=80%3A10942&mode=dev).

This PR is currently pointing at the `election-candidate-collection` branch, but once that's merged this should go straight into the `ea-giving-season-23` feature branch.

Screenshot 2023-10-26 at 17 43 00

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205814718636852) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by oetherington** [comment]
@agnesstenlund

**Comment by agnesstenlund** [comment]
looks great 👏  🚢

File: packages/lesswrong/components/ea-forum/giving-portal/PreVoteButton.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  const onVote = useCallback(() => {
+    if (currentUser) {
+      vote({
+        document,
+        voteType: null,
+        extendedVote: {preVote: !hasVoted},
+        currentUser,
+      });
+    } else {
+      window.location.href = `/auth/auth0?returnTo=${pathname}`;
```
Comment (Will-Howard) [inline packages/lesswrong/components/ea-forum/giving-portal/PreVoteButton.tsx:43]: I think we should open the login popup here instead (as that's what we do everywhere else)

File: packages/lesswrong/components/votes/SmallSideVote.tsx [hunk 2]
```diff
@@ -9,14 +9,18 @@ const SmallSideVote = ({document, hideKarma=false, collection}: {
...
-  
-  const VotingComponent = votingSystem.getCommentVotingComponent();
-  return <VotingComponent
-    document={document}
-    hideKarma={hideKarma}
-    collection={collection}
-    votingSystem={votingSystem}
-  />
+
+  const VotingComponent = votingSystem.getCommentVotingComponent?.();
```
Comment (Will-Howard) [inline packages/lesswrong/components/votes/SmallSideVote.tsx:13]: Under what circumstances will this be null?

**Review by Will-Howard — COMMENTED** [review]
The actual voting isn't working for me, it registers briefly but then disappears. Lmk if I'm doing something wrong (e.g. if there is a migration I should run):

https://github.com/ForumMagnum/ForumMagnum/assets/77623106/a4bdb548-3b85-4e92-9be6-1cbc99105714



Also, unrelated to this PR but just thought I'd flag it here. The timeline doesn't look great on dark mode:
Screenshot 2023-10-27 at 21 39 29
Everything else in dark mode looks good 👍

File: packages/lesswrong/components/votes/SmallSideVote.tsx (hunk 2 above)
Comment (oetherington) [reply packages/lesswrong/components/votes/SmallSideVote.tsx:13]: If somebody tried to use the election candidate voting system for a post (which is possible in theory, but not given as an option by the UI).

**Pull request merged**